src/sql/src/plan/expr.rs

```diff
@@ -1313,7 +1313,7 @@ impl HirScalarExpr {
     /// Column references in parameters will be corrected to account for the
     /// depth at which they are spliced.
     pub fn splice_parameters(&mut self, params: &[HirScalarExpr], depth: usize) {
-        let _ = self.visit_recursively_mut(depth, &mut |_: usize,
```

Comment:
The original code before the refactoring was:

```
    pub fn splice_parameters(&mut self, params: &[HirScalarExpr], depth: usize) {
        self.visit_mut(&mut |e| match e {
            HirScalarExpr::Parameter(i) => {
                *e = params[*i - 1].clone();
                // Correct any column references in the parameter expression for
                // its new depth.
                e.visit_columns_mut(0, &mut |d, col| {
                    if col.level >= d {
                        col.level += depth
                    }
                });
            }
            HirScalarExpr::Exists(e) | HirScalarExpr::Select(e) => {
                e.splice_parameters(params, depth + 1)
            }
            _ => (),
        })
    }
```

But now the depth increment is done by `visit_recursively_mut`, but we were ignoring it.